src/gerbil/prelude/core.ss

```diff
@@ -1707,27 +1713,24 @@ package: gerbil
                           (type-name
                            (or (stx-getq name: #'rest)
                                #'type-t))
-                          (type-plist
-                           (or (stx-getq plist: #'rest)
+                          (type-alist
+                           (or (stx-getq alist: #'rest)
                                #'[]))
-                          (type-ctor
+                          (type-constructor
                            (stx-getq constructor: #'rest))
+                          ((slot ...)
+                           (stx-map slot-name slots))
                           (type-rtd
                            (if struct?
-                             (with-syntax ((fields (stx-length els))
-                                           ((field-id ...)
-                                            (stx-map field-id els)))
+                             (with-syntax ((fields (stx-length slots)))
```

Comment:
is this actually safe? What about duplicate (merged) fields/slots from super?

Reply:
The more I think about this, the more I think it is problematic with merged slots. We might be fine if we have the implicit "don't do dupes" requirement, but maybe we should rework it to do flat repr of all the slots/fields?